### ISSUE TITLE
ESR 45: Wrong menu separators

### ISSUE THREAD

**Issue by dmik** [bitwiseworks/mozilla-os2#204]
ESR 45 menus (main menu and popups) use sunken boxes instead of lines as separators of menu entries. This is how it looks in:

* ESR 31:
![2017-05-09 1 27 43](https://cloud.githubusercontent.com/assets/900726/25828266/f814362a-3457-11e7-91e9-1b3917e4aa68.png)
* ESR 38:
![2017-05-09 1 25 35](https://cloud.githubusercontent.com/assets/900726/25828270/fd1963b6-3457-11e7-9fcf-149cf429bf05.png)
* ESR 45:
![2017-05-09 1 29 44](https://cloud.githubusercontent.com/assets/900726/25828275/036118d6-3458-11e7-95cb-ec25bb03ac92.png)

I believe proper separators got lost when syncing some XUL/CSS updates. Not easy to find though.

PS. Note that ESR 45 fixes vertical text alignment within the menu item rectangle which was broken in ESR 38 (text was stuck to the top of the rectangle).

**Comment by dmik** [comment]
Also the menu border looks different in ESR 45: clearly not OS/2 style. Perhaps we fall back to some Motif look and feel for some reason.

**Comment by dryeo** [comment]
I see the same on TB and SM, so not a theme issue. Hmm, actually I just 
changed themes from classic to modern (both are always built) and the 
sunken separators are now lines as well the menu border looks normal.
Don't know what they look like on Windows but as we mostly use the 
Windows themes...
There's going to be a lot of breakages as Mozilla moves away from XUL 
and XPCOM.

**Comment by SilvanScherrer** [comment]
IMHO the separators look and feel is not the biggest issue. But not having a arrow for submenus is more important.

**Comment by dmik** [comment]
This all comes from one place I'm sure.

**Comment by dmik** [comment]
Gosh, `toolkit/themes/os2/global/jar.mn` got broken in fc1fe16f, some weird merge failure that snuck in not undone. Restoring this file fixes the problem. But while at this I also want to fix this vertical font alignment that annoys me now

**Comment by dmik** [comment]
Besides fixing the main issue, I also fixed several other visual glitches (including the annoying vertical text misalignment), now menus look like this:

![2017-05-10 5 39 29](https://cloud.githubusercontent.com/assets/900726/25881075/a058736e-3543-11e7-9200-a7fbda71aa3c.png)

![2017-05-10 5 40 45](https://cloud.githubusercontent.com/assets/900726/25881076/a4202ab4-3543-11e7-8a5e-ba2a8000175a.png)

This is much more native than before (compare to a native PM app).

**Comment by dmik** [comment]
JFTR. Please note the following improvements (all mimic native popup behaviour):

- There is now a complete dark border around the popup (it used to be white on top at least in Firefox 10 and may even earlier due to some fancy hack made by someone long-long ago).
- The submenu in the main menu is shifted left by some pixels and sticks to the blue bar of the selected menu (i.e. no 1px space there any more).
- The blue selection bar width matches the native width (18px) and has text properly centered vertically.
- The icons are centered with text.

**Comment by dmik** [comment]
I have to reopen this, because after a7318a7 `mach build-backend` (part of configure) fails with this:
```
Reticulating splines...
Traceback (most recent call last):
  File "./config.status", line 1010, in <module>
    config_status(**args)
  File "D:/Coding/mozilla/master/python/mozbuild/mozbuild/config_status.py", line 178, in config_status
    the_backend.consume(definitions)
  File "D:/Coding/mozilla/master/python/mozbuild/mozbuild/backend/base.py", line 120, in consume
    if not self.consume_object(obj):
  File "D:/Coding/mozilla/master/python/mozbuild/mozbuild/backend/fastermake.py", line 75, in consume_object
    self._consume_jar_manifest(obj, defines)
  File "D:/Coding/mozilla/master/python/mozbuild/mozbuild/backend/fastermake.py", line 207, in _consume_jar_manifest
    mozpath.join(jarinfo.name, e.output))
  File "D:/Coding/mozilla/master/python/mozbuild/mozpack/manifests.py", line 243, in add_symlink
    self._add_entry(dest, (self.SYMLINK, source))
  File "D:/Coding/mozilla/master/python/mozbuild/mozpack/manifests.py", line 309, in _add_entry
    raise ValueError('Item already in manifest: %s' % dest)
ValueError: Item already in manifest: chrome/toolkit/skin/classic/global/popup.css
```
In fact, both Windows and OS/2 jar.mn list popup.css now and the build framework can't stand that any longer. That's why I removed it back then. I need to find another way to bring this in.

**Comment by dmik** [comment]
Fixed in 64d3a41b6d39ab82e169b144c32b6f2815d78a6f.
